applications/RANSApplication/custom_conditions/incompressible_potential_flow/incompressible_potential_flow_pressure_body_force_condition.cpp

```diff
@@ -0,0 +1,230 @@
+{
+    if (ConditionDofList.size() != TNumNodes) {
+        ConditionDofList.resize(TNumNodes);
+    }
+
+    for (unsigned int i = 0; i < TNumNodes; ++i) {
+        ConditionDofList[i] = Condition::GetGeometry()[i].pGetDof(PRESSURE_POTENTIAL);
+    }
+}
+
+template <unsigned int TDim, unsigned int TNumNodes>
+void IncompressiblePotentialFlowPressureBodyForceCondition<TDim, TNumNodes>::GetValuesVector(
+    VectorType& rValues,
+    int Step) const
+{
+    if (rValues.size() != TNumNodes) {
+        rValues.resize(TNumNodes, false);
+    }
+
+    const auto& r_geometry = this->GetGeometry();
+    IndexType local_index = 0;
+    for (IndexType i_node = 0; i_node < TNumNodes; ++i_node) {
+        rValues[local_index++] =
+            r_geometry[i_node].FastGetSolutionStepValue(PRESSURE_POTENTIAL, Step);
+    }
+}
+
+template <unsigned int TDim, unsigned int TNumNodes>
+GeometryData::IntegrationMethod IncompressiblePotentialFlowPressureBodyForceCondition<TDim, TNumNodes>::GetIntegrationMethod()
+{
+    return GeometryData::GI_GAUSS_1;
+}
+
+template <unsigned int TDim, unsigned int TNumNodes>
+void IncompressiblePotentialFlowPressureBodyForceCondition<TDim, TNumNodes>::CalculateLocalSystem(
+    MatrixType& rLeftHandSideMatrix,
+    VectorType& rRightHandSideVector,
+    const ProcessInfo& rCurrentProcessInfo)
+{
+    // Calculate RHS
+    this->CalculateRightHandSide(rRightHandSideVector, rCurrentProcessInfo);
+
+    // Calculate LHS
+    this->CalculateLeftHandSide(rLeftHandSideMatrix, rCurrentProcessInfo);
+}
+
+template <unsigned int TDim, unsigned int TNumNodes>
+void IncompressiblePotentialFlowPressureBodyForceCondition<TDim, TNumNodes>::CalculateLeftHandSide(
+    MatrixType& rLeftHandSideMatrix,
+    const ProcessInfo& rCurrentProcessInfo)
+{
+    KRATOS_TRY
+
+    if (rLeftHandSideMatrix.size1() != TNumNodes || rLeftHandSideMatrix.size2() != TNumNodes) {
+        rLeftHandSideMatrix.resize(TNumNodes, TNumNodes, false);
+    }
+
+    noalias(rLeftHandSideMatrix) = ZeroMatrix(TNumNodes, TNumNodes);
+
+    KRATOS_CATCH("");
+}
+
+template <unsigned int TDim, unsigned int TNumNodes>
+void IncompressiblePotentialFlowPressureBodyForceCondition<TDim, TNumNodes>::CalculateRightHandSide(
+    VectorType& rRightHandSideVector,
+    const ProcessInfo& rCurrentProcessInfo)
+{
+    KRATOS_TRY
+
+    if (rRightHandSideVector.size() != TNumNodes) {
+        rRightHandSideVector.resize(TNumNodes, false);
+    }
+
+    noalias(rRightHandSideVector) = ZeroVector(TNumNodes);
+
+    const auto& r_geometry = this->GetGeometry();
+    // Get Shape function data
+    Vector gauss_weights;
+    Matrix shape_functions;
+    RansCalculationUtilities::CalculateConditionGeometryData(
+        r_geometry, this->GetIntegrationMethod(), gauss_weights, shape_functions);
+    const IndexType num_gauss_points = gauss_weights.size();
+
+    array_1d<double, 3> r_normal = this->GetValue(NORMAL);
+    r_normal = r_normal * (1.0 / norm_2(r_normal));
```

Comment:
```suggestion
    r_normal /= norm_2(r_normal));
```
Besides this I'd check  the norm (although it is already checked in the `Check` method).

Reply:
done :)